src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,198 @@
+                                                                    String initialDate, String finalDate)
+            throws DatabaseAccessException {
+        CollectionReference medsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        try {
+            getMeadicationsBetweenDatesFromDatabase(initialDate, finalDate, medsRef, externalList);
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        return externalList;
+    }
+
+    private void getMeadicationsBetweenDatesFromDatabase(String initialDate, String finalDate,
+                                                         CollectionReference medsRef,
+                                                         List<Map<List<String>, Object>> externalList)
+            throws InterruptedException, ExecutionException {
+        List<QueryDocumentSnapshot> medDocuments = medsRef.get().get().getDocuments();
+        for (QueryDocumentSnapshot medDocument : medDocuments) {
+            List<String> aux = pkToList(medDocument.getId());
+            if (initialDate.compareTo(pkToDate(medDocument.getId())) < 0
+                    && finalDate.compareTo(pkToDate(medDocument.getId())) > 0) {
+                Map<List<String>, Object> internalList = new HashMap<>();
+                internalList.put(Collections.singletonList(DATENAME), aux);
+                internalList.put(Collections.singletonList(BODY),
+                        medDocument.toObject(MedicationEntity.class));
+                externalList.add(internalList);
+            }
+        }
+    }
+
+    @Override
+    public Object getMedicationField(String owner, String petName, String dateName, String field)
+            throws DatabaseAccessException {
+        ApiFuture<DocumentSnapshot> future = getMedicationsRef(owner, petName).document(dateName).get();
+        DocumentSnapshot medicationDoc;
+        try {
+            medicationDoc = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        if (!medicationDoc.exists()) {
+            throw new DatabaseAccessException(INVALID_MEDICATION_EXC,
+                    MEDICATION_DOES_NOT_EXIST_EXC);
+        }
+        return medicationDoc.get(field);
+    }
+
+    @Override
+    public void updateMedicationField(String owner, String petName, String dateName, String field,
+                                      Object value) {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        DocumentReference medRef = medicationsRef.document(dateName);
+        medRef.update(field, value);
+    }
+
+
+    public CollectionReference getMedicationsRef(String owner, String petName) {
```

Comment:
What happens if the user, the pet or both don't exist?

Reply:
Qué cambio me estás pidiendo que ponga aquí exactamente...? No te entiendo, lo siento.